### PR TITLE
Introduce Zoom Meeting Component XBlock [EDLY-6408]

### PR DESCRIPTION
**Pull request by mfarhan943**
**Description:**
Introducing Zoom Meeting Component XBlock, integrating Zoom meeting functionalities directly into the course environment. This enhancement enables users to easily participate in Zoom meetings from within their learning platform, streamlining the process and enriching the educational experience. 

**Jira**
https://edlyio.atlassian.net/browse/EDLY-6408

### CODE REVIEW

**Review by ziafazal — CHANGES_REQUESTED** [review]
@mfarhan943 I have few suggestions.

File: zoom_meeting/zoom_meeting.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        """
+        The primary view of the ZoomMeetingXBlock, shown to students
+        when viewing courses.
+        """
+        html = self.resource_string("static/html/zoom_meeting.html")
+        frag = Fragment(html.format(self=self))
+        frag.add_css(self.resource_string("static/css/zoom_meeting.css"))
+
+        js_urls = [
+            "public/lib/react.min.js",
```
[the remark this answers is not in the document]
Comment (mfarhan943) [reply zoom_meeting/zoom_meeting.py:118]: public dir have been added.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                data.append(os.path.relpath(os.path.join(dirname, fname), pkg))
+
+    return {pkg: data}
+
+
+setup(
+    name='zoom_meeting-xblock',
+    version='0.1',
+    description='zoom_meeting XBlock',   # TODO: write a better description.
+    license='UNKNOWN',          # TODO: choose a license: 'AGPL v3' and 'Apache 2.0' are popular.
```
[the remark this answers is not in the document]
Comment (mfarhan943) [reply setup.py]: Updated

File: README.md [hunk 3]
```diff
@@ -1 +1,32 @@
...
+### Install Xblock
+
+```
+pip install -e zoom_xblock
+```
+
+### **Update  Settings of LMS and CMS**
+Add the following values in the lms and cms configuration file
+```python
+ZOOM_MEETING_SDK_KEY: 'zoom-sdk-key or zoom-client-id'
```
[the remark this answers is not in the document]
Comment (mfarhan943) [reply README.md]: Removed.

File: zoom_meeting/zoom_meeting.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        payload = {
+            "sdkKey": self.sdk_key,
+            "mn": self.meeting_number,
+            "role": role,
+            "iat": iat,
+            "exp": exp,
+            "appKey": self.sdk_key,
+            "tokenExp": exp,
+        }
+        token = jwt.encode(payload, self.sdk_secret, algorithm="HS256")
```
Comment (ziafazal) [inline zoom_meeting/zoom_meeting.py:106]: can we pick this from `JWT_ALGORITHM` key of `JWT_AUTH` setting?

File: zoom_meeting/static/js/src/zoom_meeting.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+          zak: config.zak_token,
+        });
+      })
+      .catch((error) => {
+        displayError(error);
+      });
+  }
+
+  function displayError(error) {
+    var errorDiv = document.createElement("div");
```
Comment (ziafazal) [inline zoom_meeting/static/js/src/zoom_meeting.js:31]: We are using vanila JS and jQuery both in this javascript file. I think we should stick to jquery API only. so far instance here we can create div like this `var errorDiv = $('<div></div>')`

same thing apply to this `document.body.appendChild(errorDiv);` line which can be rewritten like this `$('body').append(errorDiv)`

**Review by ziafazal — CHANGES_REQUESTED** [review]
@mfarhan943 completed second round of review.

File: zoom_meeting/zoom_meeting.py [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+        ]
+
+        for js_url in js_urls:
+            frag.add_javascript_url(self.runtime.local_resource_url(self, js_url))
+
+        frag.add_javascript(self.resource_string("static/js/src/zoom_meeting.js"))
+        frag.initialize_js("ZoomMeetingXBlock", self.get_congfiurations())
+        return frag
+
+    def author_view(self, context=None):
```
[the remark this answers is not in the document]
Comment (mfarhan943) [reply zoom_meeting/zoom_meeting.py:139]: The libraries used by zoom aren't correctly loading in studio, so it is functional only in LMS view.

File: zoom_meeting/static/js/src/zoom_meeting.js (hunk 5 above)
Comment (mfarhan943) [reply zoom_meeting/static/js/src/zoom_meeting.js]: Done

File: zoom_meeting/static/js/src/zoom_meeting.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+  function getOrCreateMeetingDiv() {
+    var meetingSDKElement = $("#meetingSDKElement");
+
+    if (!meetingSDKElement.length) {
+        meetingSDKElement = $("<div></div>");
+        meetingSDKElement.attr("id", "meetingSDKElement");
+        $("body").append(meetingSDKElement);
+    }
+    console.log('zxaa ', meetingSDKElement)
```
Comment (ziafazal) [inline zoom_meeting/static/js/src/zoom_meeting.js:46]: what is purpose of this console log?

**Review by ziafazal — APPROVED** [review]
LGTM 👍

**Pull request merged**